lgc/util/TypeLowering.cpp

```diff
@@ -208,15 +209,19 @@ ArrayRef<Value *> TypeLowering::getValueOptional(Value *value) {
     assert(valueIt != m_valueMap.end());
   }
 
-  if ((valueIt->second & 1) == 0)
-    return reinterpret_cast<Value *>(valueIt->second);
+  if ((valueIt->second & 1) == 0) {
+    remapped.push_back(reinterpret_cast<Value *>(valueIt->second));
+    return remapped;
+  }
 
   size_t begin = valueIt->second >> 1;
   auto typeIt = m_multiTypeConversions.find(value->getType());
   assert(typeIt != m_multiTypeConversions.end());
   size_t count = typeIt->second.size();
+  for (unsigned v = begin; v < begin + count; v++)
+    remapped.push_back(m_convertedValueList[v]);
```

Comment:
SmallVector has `append(begin, end)`. Maybe `remapped.append(&m_convertedValueList[begin], &m_convertedValueList[begin + count])` works (not quite sure if it works with pointers)?

Reply:
We can also write like: `return SmallVector<Value *>(ArrayRef(&m_convertedValueList[begin], count));`, Not sure which is preferred?

Reply:
I’m fine with either way :)